oap-server/server-fetcher-plugin/prometheus-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/fetcher/prometheus/provider/PrometheusFetcherProvider.java

```diff
@@ -0,0 +1,87 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.fetcher.prometheus.provider;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import org.apache.skywalking.oap.server.core.CoreModule;
+import org.apache.skywalking.oap.server.core.analysis.TimeBucket;
+import org.apache.skywalking.oap.server.core.analysis.meter.MeterEntity;
+import org.apache.skywalking.oap.server.core.analysis.meter.MeterSystem;
+import org.apache.skywalking.oap.server.core.analysis.meter.ScopeType;
+import org.apache.skywalking.oap.server.core.analysis.meter.function.AcceptableValue;
+import org.apache.skywalking.oap.server.fetcher.prometheus.module.PrometheusFetcherModule;
+import org.apache.skywalking.oap.server.library.module.ModuleConfig;
+import org.apache.skywalking.oap.server.library.module.ModuleDefine;
+import org.apache.skywalking.oap.server.library.module.ModuleProvider;
+import org.apache.skywalking.oap.server.library.module.ModuleStartException;
+import org.apache.skywalking.oap.server.library.module.ServiceNotProvidedException;
+
+public class PrometheusFetcherProvider extends ModuleProvider {
+    private final PrometheusFetcherConfig config;
+
+    public PrometheusFetcherProvider() {
+        config = new PrometheusFetcherConfig();
+    }
+
+    @Override
+    public String name() {
+        return "default";
+    }
+
+    @Override
+    public Class<? extends ModuleDefine> module() {
+        return PrometheusFetcherModule.class;
+    }
+
+    @Override
+    public ModuleConfig createConfigBeanIfAbsent() {
+        return config;
+    }
+
+    @Override
+    public void prepare() throws ServiceNotProvidedException, ModuleStartException {
+        final MeterSystem meterSystem = MeterSystem.meterSystem(getManager());
+        meterSystem.create("test_long_metrics", "avg", ScopeType.SERVICE, Long.class);
+    }
+
+    @Override
+    public void start() throws ServiceNotProvidedException, ModuleStartException {
+
+    }
+
+    @Override
+    public void notifyAfterCompleted() throws ServiceNotProvidedException, ModuleStartException {
+        final MeterSystem service = getManager().find(CoreModule.NAME).provider().getService(MeterSystem.class);
+        Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                final AcceptableValue<Long> value = service.buildMetrics("test_long_metrics", Long.class);
+                value.accept(MeterEntity.newService("abc"), 5L);
+                value.setTimeBucket(TimeBucket.getMinuteTimeBucket(System.currentTimeMillis()));
+                service.doStreamingCalculation(value);
+            }
+        }, 2, 2, TimeUnit.SECONDS);
+    }
```

Comment:
Is FetchProvider a demonstration?
The Metrics are reported by Prometheus client or SkyWalking Agent.

Reply:
For demo, yes. Prometheus doesn't have client, OAP fetchs data from target instrumented applications.